ruby/sdk-lightweight.md

```diff
@@ -0,0 +1,314 @@
+
+Activities are defined as classes ([a very common practice](https://guides.rubyonrails.org/active_job_basics.html)
+given that the language is object-oriented).
+
+`sum_activity.rb`
+
+```ruby
+class SumActivity < Temporal::Activity
+  # This will generate a default class-level stub SumActivity.execute to be used in workflows
+  def execute(a, b)
+    a + b
+  end
+end
+```
+
+#### Alternative approaches
+
+- Using plain methods as activities. This is not feasible in Ruby since non-namespaced methods
+  are global without a clear way to reference them (would have to be `method(:my_activity)`)
+
+- It is possible to allow multiple activities per class, for example:
+
+`calculator_activities.rb`
+
+```ruby
+class CalculatorActivities < Temporal::Activity
+  # Generate custom stubs on the class to be used in workflows:
+  # - CalculatorActivities.add
+  # - CalculatorActivities.subtract
+  activities :add, :subtract
+
+  def add(a, b)
+    a + b
+  end
+
+  def subtract(a, b)
+    a - b
+  end
+end
+```
+
+
+## Defining a workflow
+
+Similar to activities, workflows are also represented as classes.
+
+`sum_workflow.rb`
+
+```ruby
+require 'sum_activity'
+
+class SumWorkflow < Temporal::Workflow
+  def execute(a, b)
+    # - .execute! is a shorthand for .execute().await
+    # - only positional arguments are supported (inline with other languages), keyword
+    #   arguments are reserved for options
+    SumActivity.execute!(a, b, start_to_close_timeout: 10)
+  end
+end
+```
+
+#### Alternative approaches
+
+- For the same exact reasons it is not feasible to use non-namespaced methods for workflow
+
+- Similar to activities it should be possible to define multiple workflows per class, however this
+  will not be allowed to avoid a perception of a shared state between workflows
+
+
+## Start a workflow
+
+```ruby
+require 'temporal/client'
+require 'sum_workflow'
+
+connection = Temporal::Connection.new('localhost:7233')
+client = Temporal::Client.new(connection, namespace: 'demo')
+
+# workflow is referenced via a stub, not just a class
+handle = client.start_workflow(SumWorkflow.execute(1, 2), id: 'id-1', task_queue: 'demo')
+handle.result # => 3
+```
+
+#### Alternative approaches
+
+- Starting a foreign workflow from a string name
+
+```ruby
+client.start_workflow(Temporal::Workflow.execute('my.workflow', 1, 2), id: 'id-1', task_queue: 'demo')
+```
+
+- Starting a foreign workflow with the help of a stub
+
+```ruby
+class MyWorkflow < Temporal::Workflow
+  workflow 'my.workflow'
+end
+
+client.start_workflow(MyWorkflow.execute('my.workflow', 1, 2), id: 'id-1', task_queue: 'demo')
+```
+
+
+## Start a worker
+
+```ruby
+require 'temporal/worker'
+require 'sum_workflow'
+require 'sum_activity'
+require 'calculator_activities'
+
+connection = Temporal::Connection.new('localhost:7233')
+worker = Temporal::Worker.new(
+  connection,
+  namespace: 'demo',
+  task_queue: 'demo',
+  workflows: [SumWorkflow], # registers all workflows in the class
```

Comment:
```suggestion
  workflows: [SumWorkflow], # registers each class as a workflow
```

Because, after discussion, a workflow class can only have one workflow. And, after our discussion, we decided to always require the entrypoint to be `execute` instead of allowing that to be customized since it's inline w/ normal Ruby nomenclature and there's no benefit in customizing it.